### DISCUSSION TITLE
repo sync ERROR downloading, [Errno 14] HTTPS Error 302 - Found

### DISCUSSION THREAD

**Discussion by ipanmeri** [uyuni-project/uyuni#8769] [Q&A]
Hi, yesterday and today seems I cannot get a complete repo sync for
ubuntu-2004-amd64-uyuni-client

I get errors like
`2024/05/16 17:44:20 +02:00 ERROR: Download failed: https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/all/salt-common_3006.0+ds-1+221.1.uyuni_all.deb - [Errno 14] HTTPS Error 302 - Found.
2024/05/16 17:44:20 +02:00 ERROR: Download failed: https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/all/spacecmd_5.0.6-2.1.uyuni_all.deb - [Errno 14] HTTPS Error 302 - Found.
2024/05/16 17:44:20 +02:00 WARNING: Download failed: https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/amd64/venv-salt-minion_3006.0-26.3.uyuni_amd64.deb - [Errno 14] HTTPS Error 302 - Found. Retrying...
2024/05/16 17:44:20 +02:00 ERROR: Download failed: https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/amd64/venv-salt-minion_3006.0-26.3.uyuni_amd64.deb - [Errno 14] HTTPS Error 302 - Found.`

Thus I can't get a bootstrap repo done.
What are these errors, how can I fix that?
It is not a problem of our firewall as I synced before with success, and other repos synced with success. Is this a temporary issue?

Thanks

**Comment by ipanmeri** [comment]
So, this is the list of all errors:

`uyunisrv:/var/log/rhn/reposync # grep -i error ubuntu-2004-amd64-uyuni-client.log
2024/05/16 17:44:20 +02:00 WARNING: Download failed: https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/all/spacecmd_5.0.6-2.1.uyuni_all.deb - [Errno 14] HTTPS Error 302 - Found. Retrying...
2024/05/16 17:44:20 +02:00 WARNING: Download failed: https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/all/salt-common_3006.0+ds-1+221.1.uyuni_all.deb - [Errno 14] HTTPS Error 302 - Found. Retrying...
2024/05/16 17:44:20 +02:00 WARNING: Download failed: https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/all/salt-common_3006.0+ds-1+221.1.uyuni_all.deb - [Errno 14] HTTPS Error 302 - Found. Retrying...
2024/05/16 17:44:20 +02:00 WARNING: Download failed: https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/all/spacecmd_5.0.6-2.1.uyuni_all.deb - [Errno 14] HTTPS Error 302 - Found. Retrying...
2024/05/16 17:44:20 +02:00 WARNING: Download failed: https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/amd64/venv-salt-minion_3006.0-26.3.uyuni_amd64.deb - [Errno 14] HTTPS Error 302 - Found. Retrying...
2024/05/16 17:44:20 +02:00 ERROR: Download failed: https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/all/salt-common_3006.0+ds-1+221.1.uyuni_all.deb - [Errno 14] HTTPS Error 302 - Found.
2024/05/16 17:44:20 +02:00 ERROR: Download failed: https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/all/spacecmd_5.0.6-2.1.uyuni_all.deb - [Errno 14] HTTPS Error 302 - Found.
2024/05/16 17:44:20 +02:00 WARNING: Download failed: https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/amd64/venv-salt-minion_3006.0-26.3.uyuni_amd64.deb - [Errno 14] HTTPS Error 302 - Found. Retrying...
2024/05/16 17:44:20 +02:00 ERROR: Download failed: https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/amd64/venv-salt-minion_3006.0-26.3.uyuni_amd64.deb - [Errno 14] HTTPS Error 302 - Found.
uyunisrv:/var/log/rhn/reposync #
`

What I tried next was to remove all the packages in the repository, so that the sync action should redownload all of them.
Packages were downloaded, but I still have the errors listed above.
I find it strange though, that although I get the
`2024/05/16 17:44:20 +02:00 ERROR: Download failed: https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/amd64/venv-salt-minion_3006.0-26.3.uyuni_amd64.deb - [Errno 14] HTTPS Error 302 - Found.`
the package is listed in the repository.

The attempt to create the bootstrap repo fails
`uyunisrv:/var/log/rhn/reposync # mgr-create-bootstrap-repo -c ubuntu-20.04-amd64-uyuni

Creating bootstrap repo for ubuntu-20.04-amd64-uyuni
copy 'libnorm1-1.5.8+dfsg2-2build1.amd64-deb'
copy 'libpgm-5.2-0-5.2.122~dfsg-3ubuntu1.amd64-deb'
copy 'libzmq5-4.3.2-2ubuntu1.amd64-deb'
copy 'python3-crypto-2.6.1-13ubuntu2.amd64-deb'
copy 'python3-zmq-18.1.1-3.amd64-deb'
copy 'python3-gnupg-0.4.5-2.all-deb'
copy 'python3-looseversion-1.0.2-2.all-deb'
copy 'salt-common-3006.0+ds-1+221.1.uyuni.all-deb'
copy 'salt-minion-3006.0+ds-1+221.1.uyuni.all-deb'
copy 'venv-salt-minion-3006.0-26.3.uyuni.amd64-deb'
Skipping inclusion of 'python3-crypto' '2.6.1-13ubuntu2' in 'bootstrap|main|amd64', as this version already exists.
Skipping inclusion of 'libnorm1' '1.5.8+dfsg2-2build1' in 'bootstrap|main|amd64', as this version already exists.
Skipping inclusion of 'python3-looseversion' '1.0.2-2' in 'bootstrap|main|amd64', as this version already exists.
Skipping inclusion of 'python3-looseversion' '1.0.2-2' in 'bootstrap|main|i386', as this version already exists.
Skipping inclusion of 'python3-looseversion' '1.0.2-2' in 'bootstrap|main|armhf', as this version already exists.
Skipping inclusion of 'python3-looseversion' '1.0.2-2' in 'bootstrap|main|arm64', as this version already exists.
Skipping inclusion of 'libpgm-5.2-0' '5.2.122~dfsg-3ubuntu1' in 'bootstrap|main|amd64', as this version already exists.
Skipping inclusion of 'salt-common' '3006.0+ds-1+221.1.uyuni' in 'bootstrap|main|amd64', as this version already exists.
Skipping inclusion of 'salt-common' '3006.0+ds-1+221.1.uyuni' in 'bootstrap|main|i386', as this version already exists.
Skipping inclusion of 'salt-common' '3006.0+ds-1+221.1.uyuni' in 'bootstrap|main|armhf', as this version already exists.
Skipping inclusion of 'salt-common' '3006.0+ds-1+221.1.uyuni' in 'bootstrap|main|arm64', as this version already exists.
Skipping inclusion of 'salt-minion' '3006.0+ds-1+221.1.uyuni' in 'bootstrap|main|amd64', as this version already exists.
Skipping inclusion of 'salt-minion' '3006.0+ds-1+221.1.uyuni' in 'bootstrap|main|i386', as this version already exists.
Skipping inclusion of 'salt-minion' '3006.0+ds-1+221.1.uyuni' in 'bootstrap|main|armhf', as this version already exists.
Skipping inclusion of 'salt-minion' '3006.0+ds-1+221.1.uyuni' in 'bootstrap|main|arm64', as this version already exists.
Skipping inclusion of 'python3-zmq' '18.1.1-3' in 'bootstrap|main|amd64', as this version already exists.
Skipping inclusion of 'venv-salt-minion' '3006.0-26.3.uyuni' in 'bootstrap|main|amd64', as this version already exists.
Skipping inclusion of 'python3-gnupg' '0.4.5-2' in 'bootstrap|main|amd64', as this version already exists.
Skipping inclusion of 'python3-gnupg' '0.4.5-2' in 'bootstrap|main|i386', as this version already exists.
Skipping inclusion of 'python3-gnupg' '0.4.5-2' in 'bootstrap|main|armhf', as this version already exists.
Skipping inclusion of 'python3-gnupg' '0.4.5-2' in 'bootstrap|main|arm64', as this version already exists.
Skipping inclusion of 'libzmq5' '4.3.2-2ubuntu1' in 'bootstrap|main|amd64', as this version already exists.
Exporting indices...
ERROR: package 'dctrl-tools' not found
ERROR: package 'python3-dateutil' not found
ERROR: package 'python3-distro' not found
ERROR: package 'python3-jinja2' not found
ERROR: package 'python3-markupsafe' not found
ERROR: package 'python3-msgpack' not found
ERROR: package 'python3-psutil' not found
ERROR: package 'python3-pycryptodome' not found
ERROR: package 'python3-jmespath' not found
ERROR: package 'gnupg' not found

Suggestions:
mgr-create-bootstrap-repo uses the locally synchronized versions of files
from the Tools repository, and uses the locally synchronized pool channel
for dependency resolution.
Both should be fully synced before running the mgr-create-bootstrap-repo script.

uyunisrv:/var/log/rhn/reposync #
`

I see no download errors for the e.g. python3 packages, but looks like they are not available to create the bootstrap.
Or is that a completely different problem?
Thanks for any advice.
Kind regards

**Reply by mcalmer** [reply]
Seems it redirect you to a local mirror, but this mirror does not have the package.
Try to call the URL with `curl -vvv -L -O https://....` and check the header fields.

**Reply by ipanmeri** [reply]
Hi and thanks for your reply.
Like so:
`uyunisrv:/var/log/rhn/reposync # curl -vvv -L -O https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/all/spacecmd_5.0.6-2.1.uyuni_all.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 195.135.223.226:443...
* Connected to download.opensuse.org (195.135.223.226) port 443 (#0)
* ALPN: offers h2,http/1.1
} [5 bytes data]
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
} [512 bytes data]
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* TLSv1.3 (IN), TLS handshake, Server hello (2):
{ [122 bytes data]
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
{ [19 bytes data]
* TLSv1.3 (IN), TLS handshake, Certificate (11):
{ [2640 bytes data]
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
{ [111 bytes data]
* TLSv1.3 (IN), TLS handshake, Finished (20):
{ [36 bytes data]
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
} [1 bytes data]
* TLSv1.3 (OUT), TLS handshake, Finished (20):
} [36 bytes data]
* SSL connection using TLSv1.3 / TLS_CHACHA20_POLY1305_SHA256
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=mirrorcache.opensuse.org
*  start date: Apr 22 00:21:12 2024 GMT
*  expire date: Jul 21 00:21:11 2024 GMT
*  subjectAltName: host "download.opensuse.org" matched cert's "download.opensuse.org"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
} [5 bytes data]
* using HTTP/2
* h2h3 [:method: GET]
* h2h3 [:path: /repositories/systemsmanagement:/Uyuni:/Stable:/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/all/spacecmd_5.0.6-2.1.uyuni_all.deb]
* h2h3 [:scheme: https]
* h2h3 [:authority: download.opensuse.org]
* h2h3 [user-agent: curl/8.0.1]
* h2h3 [accept: */*]
* Using Stream ID: 1 (easy handle 0x555e32af63e0)
} [5 bytes data]
> GET /repositories/systemsmanagement:/Uyuni:/Stable:/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/all/spacecmd_5.0.6-2.1.uyuni_all.deb HTTP/2
> Host: download.opensuse.org
> user-agent: curl/8.0.1
> accept: */*
>
{ [5 bytes data]
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
{ [57 bytes data]
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
{ [57 bytes data]
* old SSL session ID is stale, removing
{ [5 bytes data]
< HTTP/2 302
< date: Fri, 17 May 2024 07:14:13 GMT
< server: Mojolicious (Perl)
< cache-control: public, max-age=300
< content-length: 0
< location: https://ftp.gwdg.de/pub/opensuse/repositories/systemsmanagement%3A/Uyuni%3A/Stable%3A/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/all/spacecmd_5.0.6-2.1.uyuni_all.deb
< vary: Accept,COUNTRY,X-COUNTRY,Fastly-SSL
< content-type: application/x-debian-package
<
{ [0 bytes data]
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
* Connection #0 to host download.opensuse.org left intact
* Issue another request to this URL: 'https://ftp.gwdg.de/pub/opensuse/repositories/systemsmanagement%3A/Uyuni%3A/Stable%3A/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/all/spacecmd_5.0.6-2.1.uyuni_all.deb'
*   Trying 134.76.12.6:443...
* Connected to ftp.gwdg.de (134.76.12.6) port 443 (#1)
* ALPN: offers h2,http/1.1
} [5 bytes data]
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
} [512 bytes data]
* TLSv1.3 (IN), TLS handshake, Server hello (2):
{ [108 bytes data]
* TLSv1.2 (IN), TLS handshake, Certificate (11):
{ [4867 bytes data]
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
{ [365 bytes data]
* TLSv1.2 (IN), TLS handshake, Server finished (14):
{ [4 bytes data]
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
} [102 bytes data]
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
} [1 bytes data]
* TLSv1.2 (OUT), TLS handshake, Finished (20):
} [16 bytes data]
* TLSv1.2 (IN), TLS handshake, Finished (20):
{ [16 bytes data]
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
* ALPN: server accepted http/1.1
* Server certificate:
*  subject: C=DE; ST=Niedersachsen; O=Gesell. f▒r wissenschaftliche Datenverarbeitung mbH, G▒ttingen; CN=ftp.gwdg.de
*  start date: Apr 27 00:00:00 2024 GMT
*  expire date: Apr 27 23:59:59 2025 GMT
*  subjectAltName: host "ftp.gwdg.de" matched cert's "ftp.gwdg.de"
*  issuer: C=GB; ST=Greater Manchester; L=Salford; O=Sectigo Limited; CN=Sectigo RSA Organization Validation Secure Server CA
*  SSL certificate verify ok.
* using HTTP/1.1
} [5 bytes data]
> GET /pub/opensuse/repositories/systemsmanagement%3A/Uyuni%3A/Stable%3A/Ubuntu2004-Uyuni-Client-Tools/xUbuntu_20.04/all/spacecmd_5.0.6-2.1.uyuni_all.deb HTTP/1.1
> Host: ftp.gwdg.de
> User-Agent: curl/8.0.1
> Accept: */*
>
{ [5 bytes data]
< HTTP/1.1 200 OK
< Server: nginx
< Date: Fri, 17 May 2024 07:14:14 GMT
< Content-Type: application/octet-stream
< Content-Length: 210228
< Last-Modified: Fri, 03 May 2024 12:31:46 GMT
< Connection: keep-alive
< ETag: "6634d932-33534"
< Strict-Transport-Security: max-age=63072000; includeSubDomains; preload
< X-Frame-Options: SAMEORIGIN
< X-Content-Type-Options: nosniff
< Accept-Ranges: bytes
<
{ [15998 bytes data]
100  205k  100  205k    0     0   461k      0 --:--:-- --:--:-- --:--:--  461k
* Connection #1 to host ftp.gwdg.de left intact
`

But the file is downloaded, so it exists:
`uyunisrv:/var/log/rhn/reposync # ls -l  space*
-rw-r--r-- 1 root root 210228 May 17 09:14 spacecmd_5.0.6-2.1.uyuni_all.deb`

The local mirror is GWDG's ftp.gwdg.de.

**Reply by ipanmeri** [reply]
Can I force to go to other mirrors?

**Reply by ipanmeri** [reply]
I would also assume that these packages
`ERROR: package 'dctrl-tools' not found
ERROR: package 'python3-dateutil' not found
ERROR: package 'python3-distro' not found
ERROR: package 'python3-jinja2' not found
ERROR: package 'python3-markupsafe' not found
ERROR: package 'python3-msgpack' not found
ERROR: package 'python3-psutil' not found
ERROR: package 'python3-pycryptodome' not found
ERROR: package 'python3-jmespath' not found
ERROR: package 'gnupg' not found`

are from the base channel not the uyuni client channel, right?

**Comment by ipanmeri** [comment]
Terminating the project. Sorry to have you bothered.
